backend/src/graphql.ts

```diff
@@ -301,5 +301,52 @@ schema.mutationType({
         })
       },
     })
+    t.field('updateComment', {
+      type: 'Comment',
+      args: {
+        commentId: intArg({ required: true }),
+        body: stringArg({ required: true }),
+      },
+      resolve: async (parent, args, ctx) => {
+        const { userId } = ctx.request
+        if (!userId) throw new Error('You must be logged in to do that.')
+
+        const comment = await ctx.db.comment.update({
+          data: {
+            body: args.body,
+          },
+          where: {
+            id: args.commentId,
+          },
```

Comment:
```suggestion
          where: {
            id: args.commentId,
            authorId: userId
          },
```

Reply:
This doesn't currently work due to `CommentWhereUniqueInput` only accepting `id` currently, so I have left a comment so we can investigate that separately.

Reply:
Hmm, if it doesn't let use filter by author ID when updating, we need to fetch the comment first and check that its authorId matches the userId from the request (or that the userId belongs to a superuser/moderator/someone who should be allowed to update the comment as opposed to just a logged in user)

Reply:
@Lanny cool that's what I tried to do originally before understanding the whole transactions support thingy. Updated this now to be more robust!

Reply:
Updated the checks on `deleteComment`, too.

Reply:
@Lanny also created a `hasPostPermissions` util function that can be used for checking permissions for creating/updating/deleting a `Post` or `Comment` 😄

Reply:
@Lanny I just added some optimizations here with `Promise.all()`